kaldb/src/main/java/com/slack/kaldb/logstore/opensearch/OpenSearchAdapter.java

```diff
@@ -379,15 +389,11 @@ private static boolean tryRegisterField(
           fieldType.familyTypeName());
       return false;
     } else {
-      try {
-        XContentBuilder mapping = fieldMapping(fieldName, buildField);
-        mapperService.merge(
-            "_doc",
-            new CompressedXContent(BytesReference.bytes(mapping)),
-            MapperService.MergeReason.MAPPING_UPDATE);
-      } catch (Exception e) {
-        LOG.error("Error doing map update", e);
-      }
+      XContentBuilder mapping = fieldMapping(fieldName, buildField);
```

Comment:
This needs to catch the case where you attempt to register the same field multiple times.